File: framework/src/play/src/main/scala/play/core/system/Invoker.scala
/*
 * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
 */
package play.core

import akka.actor._
import com.typesafe.config._
import play.api.{ Logger, Play }
import scala.concurrent.ExecutionContext

/**
 * provides Play's internal actor system and the corresponding actor instances
 */
private[play] object Invoker {

  val lazySystem = new ClosableLazy[ActorSystem] {

    private val logger = Logger(this.getClass)

    protected def create() = {
      val system = Play.maybeApplication.map { app =>
        ActorSystem("play", loadActorConfig(app.configuration.underlying), app.classloader)
      } getOrElse {
        logger.warn("No application found at invoker init")
        ActorSystem("play", loadActorConfig(ConfigFactory.load()))
      }

      val close: CloseFunction = { () =>
        system.shutdown()
        system.awaitTermination()
      }

      (system, close)
    }

    private def loadActorConfig(config: Config): Config = {
      try {
        config.getConfig("play")
      } catch {
        case _: ConfigException.Missing => ConfigFactory.empty()
      }
    }

  }

  def system: ActorSystem = lazySystem.get()
  def executionContext: ExecutionContext = system.dispatcher

}

File: framework/src/play/src/main/java/play/libs/Invoker.java
package play.libs;

import akka.actor.*;
import akka.actor.Actor.*;
import scala.concurrent.ExecutionContext;
/**
 * provides Play's internal actor system and the corresponding execution context
 */
class Invoker {

   private static final ActorSystem sys = ActorSystem.apply("play");

  /**
   * provides actor system for Java Promises
   */
  static ActorSystem system() {return sys;}
  
 /**
  * provides execution context for Java Promises
  */ 
  static ExecutionContext executionContext() {return sys.dispatcher();}
 

}
